### ISSUE TITLE
filters.csf extremely slow in linux

### ISSUE THREAD

**Issue by nmgeomatics** [PDAL/PDAL#3690]
Running filters.csf in python in linux appears to have an issue. I run a pipeline in a python script in both windows and linux with the filters.csf and the script runs in seconds in windows (windows 11), but it runs for hours in linux (ubuntu).

Snippet from the script

```
csfrigidness=2
csfthreshold=0.25

pipeline_json= f"""
[
  {{
      "type":"readers.las",
      "filename":"{infile}"
  }},
  {{
        "type":"filters.csf",
        "ignore":"Classification[7:7]",
        "rigidness":"{csfrigidness}",
        "threshold":"{csfthreshold}"
    }}
]
"""

pipeline = pdal.Pipeline(pipeline_json)
pipeline.execute()
```

**Comment by chambbj** [comment]
There should be nothing platform-dependent about `filters.csf`. I'm wondering if you are running the same version on each system. There is an unreleased change (if you are building from source) that would allow you to skip writing some intermediate outputs, which can speed things up significantly.

You'll need to let us know how you are installing PDAL, which versions you are installing, and ideally provide some sample data.

**Comment by abellgithub** [comment]
There was also this: https://github.com/PDAL/PDAL/issues/3248, but it's also not system-dependent, but if you had an older Linux version...

**Comment by abellgithub** [comment]
I'll also add that in the past we have found libstdc++ to behave differently from other C++ standard libraries and there could be a performance issue in that library that you have hit. If you could do some further diagnosis and verify versions, that would be helpful.

**Comment by nmgeomatics** [comment]
My installation is as follows:
Ubuntu 20.04.3 LTS installed in Windows Subsystem For Linux 2 on Windows 11
Using Anaconda 4.11.0 and python 3.9.10 in a anaconda virtual environment
The conda list contents are attached which will show you the pdal version installed.
[myconda.txt](https://github.com/PDAL/PDAL/files/8145311/myconda.txt)

I am not sure if I can provide the sample data since it is from a company I am doing work for. I would have to get permission for that.

However, I wonder if the intermediate file thing mentioned by @chambbj  is the issue here since I am running in WSL and I bet it is slower to write to the Windows file system which might explain why it is much quicker when running in Windows directly.
I do see those intermediate files in the file system and when I do a refresh in Windows Explorer I can see the file size slowly increasing. So I imagine this is the problem. Because of this, I will look into possibly building myself or waiting for the next version which will hopefully contain this change and hopefully won't take too long to be released.

**Comment by n4z4m3** [comment]
This issue is resolved with #3952
